src/Components/PaperEditor/ProgrammeScriptContainer/index.js

```diff
@@ -281,12 +281,64 @@ const ProgrammeScriptContainer = (props) => {
           };
           newElement.words.push(newWord);
         });
+
+        elementsClone.splice(insertElementIndex, 0, newElement);
+        const updatedElements = updateWordTimingsAfterInsert(elementsClone, insertElementIndex);
+        setElements(updatedElements);
+        setResetPreview(true);
+        handleSaveProgrammeScript(updatedElements);
+
+      } else {
+        const wordsArray = divideWordsSelectionsIntoParagraphs(result.words);
+
+        const elsToAdd = wordsArray.reduce((prevResult, array) => {
+          const elStart = prevResult.newDuration;
+          const elEnd = elStart + (array[array.length - 1].end - array[0].start);
```

Comment:
`array[array.length - 1]` and `array[0]` should be variables from a readability + reusability point of view :)